mlbench_core/dataset/linearmodels/pytorch/dataloader.py

```diff
@@ -0,0 +1,231 @@
+    {'name': 'rcv1-train', 'n_samples': 677399, 'n_features': 47236, 'sparse': True},
+    {'name': 'synthetic-dense', 'n_samples': 10000, 'n_features': 100, 'sparse': False},
+]
+
+
+class IMDBPT(torch.utils.data.Dataset):
+    """
+    Args:
+        root (string): Either root directory for the database files,
+            or a absolute path pointing to the file.
+        classes (string or list): One of {'train', 'val', 'test'} or a list of
+            categories to load. e,g. ['bedroom_train', 'church_train'].
+        transform (callable, optional): A function/transform that
+            takes in an PIL image and returns a transformed version.
+            E.g, ``transforms.RandomCrop``
+        target_transform (callable, optional):
+            A function/transform that takes in the target and transforms it.
+    """
+
+    def __init__(self, root, transform=None, target_transform=None, is_image=True, n_features=None):
+        self.n_features = n_features
+        self.root = os.path.expanduser(root)
+        self.transform = transform
+        self.target_transform = target_transform
+        self.lmdb_files = self._get_valid_lmdb_files()
+
+        # for each class, create an LSUNClassDataset
+        self.dbs = []
+        for lmdb_file in self.lmdb_files:
+            self.dbs.append(LMDBPTClass(
+                root=lmdb_file, transform=transform,
+                target_transform=target_transform, is_image=is_image))
+
+        # build up indices.
+        self.indices = np.cumsum([len(db) for db in self.dbs])
+        self.length = self.indices[-1]
+        self._get_index_zones = self._build_indices()
+
+    def _get_valid_lmdb_files(self):
+        """get valid lmdb based on given root."""
+        if not self.root.endswith('.lmdb'):
+            for l in os.listdir(self.root):
+                if '_' in l and '-lock' not in l:
+                    yield os.path.join(self.root, l)
+        else:
+            yield self.root
+
+    def _build_indices(self):
+        indices = self.indices
+        from_to_indices = enumerate(zip(indices[: -1], indices[1:]))
+
+        def f(x):
+            if len(list(from_to_indices)) == 0:
+                return 0, x
+
+            for ind, (from_index, to_index) in from_to_indices:
+                if from_index <= x and x < to_index:
+                    return ind, x - from_index
+
+        return f
+
+    def _get_matched_index(self, index):
+        return self._get_index_zones(index)
+
+    def __getitem__(self, index):
+        """
+        Args:
+            index (int): Index
+        Returns:
+            tuple: Tuple (image, target)
+        """
+        block_index, item_index = self._get_matched_index(index)
+        image, target = self.dbs[block_index][item_index]
+        return image, target
+
+    def __len__(self):
+        return self.length
+
+    def __repr__(self):
+        fmt_str = 'Dataset ' + self.__class__.__name__ + '\n'
```

Comment:
maybe use an implicitely joined string instead of the addition to make the code more readable, with a single `.format()` call
```
string = ("this is an \n"
          "implicitly joined \n"
          "string")
```